skbio/stats/composition.py

```diff
@@ -608,10 +611,269 @@ def centralize(mat):
+
+    This method can be extended to an arbitrary number of classes
+    by passing in a different statistical function.
+
+    Parameters
+    ----------
+    table : pd.DataFrame or array_like
+       A 2D matrix of counts or proportions where
+       the rows correspond to samples and the
+       columns correspond to features
+    groupings : pd.Series or 1D array_like
+       Vector of group categories
+    alpha : float, optional
+       Significance level for each of the statistical tests
+    tau : float, optional
+       A constant used to determine an appropriate
+       cutoff
+    theta : float, optional
+       Lower bound for the proportion of differential features
+    multiple_comparisons_correction : {None, 'holm-bonferroni'}, optional
+       The multiple comparison correction procedure to run
+    significance_test : function, optional
+       A statistical signficance function to test for
+       signficance between classes.
+
+    Returns
+    -------
+    pd.DataFrame
+        A table of features, their W-statistics and whether
+        the null hypothesis is rejected
+
+    References
+    ----------
+    ..[1] Mandal et al. "Analysis of composition of microbiomes:
+          a novel method for studying microbial composition",
+          Microbial Ecology in Health & Disease, (2015), 26
+
+    Examples
+    --------
+    First import all of the necessary modules
+
+    >>> from skbio.stats.composition import ancom
+    >>> import pandas as pd
+
+    Now lets load in a pandas dataframe with sample and feature ids
+    for our data matrix
+
+    >>> table = pd.DataFrame(
+    ...     [[10., 11., 10., 10., 10., 10., 10.],
+    ...      [10.5, 11.5, 10.5, 10.5, 10.5, 10.5, 10.5],
+    ...      [10., 11., 10., 10., 10., 10., 10.],
+    ...      [20., 21., 10., 10., 10., 10., 10.],
+    ...      [20.5, 21.5, 10.5, 10.5, 10.5, 10.5, 10.5],
+    ...      [20.3, 21.3, 10.2, 10.3, 10.1, 10.6, 10.4]],
+    ...     index=['s1','s2','s3','s4','s5','s6'],
+    ...     columns=['b1','b2','b3','b4','b5','b6','b7'])
+
+    Then create a create a category vector.  In this scenerio, there
+    are only two classes, so the first three samples fall under the first
+    class while the last three samples fall under the last class
+
+    >>> cats = pd.Series([0, 0, 0, 1, 1, 1],
+    ...                  index=['s1','s2','s3','s4','s5','s6'])
+
+    Now run ancom and see if there are any features that have any
+    significant differences
+
+    >>> results = ancom(table, cats)
+    >>> results['W']
+    b1    6
+    b2    6
+    b3    2
+    b4    2
+    b5    2
+    b6    2
+    b7    2
+    Name: W, dtype: int64
+    >>> results['reject']
+    b1     True
+    b2     True
+    b3    False
+    b4    False
+    b5    False
+    b6    False
+    b7    False
+    Name: reject, dtype: bool
+    """
+    if len(table) != len(groupings):
+        raise ValueError('The number of samples in mat needs' +
```

Comment:
I've noticed that the ValueErrors under `skbio.stats.distance._base._preprocess_input` don't backtick the variable names.  Would it be desirable to backtick the ValueErrors in this PR and raise an issue for those ValueErrors in `_base.py`?